### PR TITLE
native: implement StdLib and CryptoLib contracts

### LINKED ISSUE #1740: Native stdlib and cryptolib contracts
Some interops are moved there, see neo-project/neo#2298.

### PR DESCRIPTION
**Pull request by AnnaShaleva**
Close #1740

### CODE REVIEW

**Review by fyrchik — CHANGES_REQUESTED** [review]
Why are last 2 commits in this PR?

File: pkg/core/native/std.go [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+	md := newMethodAndPrice(s.serialize, 1<<12, callflag.NoneFlag)
+	s.AddMethod(md, desc)
+
+	desc = newDescriptor("deserialize", smartcontract.AnyType,
+		manifest.NewParameter("data", smartcontract.ByteArrayType))
+	md = newMethodAndPrice(s.deserialize, 1<<14, callflag.NoneFlag)
+	s.AddMethod(md, desc)
+
+	desc = newDescriptor("jsonSerialize", smartcontract.ByteArrayType,
+		manifest.NewParameter("data", smartcontract.AnyType))
```
[the remark this answers is not in the document]
Comment (AnnaShaleva) [reply pkg/core/native/std.go:50]: Fixed. I forgot about the names.

File: pkg/core/native/std.go [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+		}
+		str = strings.ToUpper(str)
+	default:
+		panic(ErrInvalidBase)
+	}
+	return stackitem.NewByteArray([]byte(str))
+}
+
+func (s *Std) atoi(_ *interop.Context, args []stackitem.Item) stackitem.Item {
+	numBytes, err := args[0].TryBytes()
```
[the remark this answers is not in the document]
Comment (AnnaShaleva) [reply pkg/core/native/std.go:181]: Fixed here and in all other places.

File: pkg/compiler/inline_test.go [hunk 3]
```diff
@@ -133,27 +133,27 @@ func TestInlineInLoop(t *testing.T) {
...
 		src := `package foo
-		import "github.com/nspcc-dev/neo-go/pkg/interop/binary"
+		import "github.com/nspcc-dev/neo-go/pkg/interop/runtime"
 		import "github.com/nspcc-dev/neo-go/pkg/compiler/testdata/inline"
 		func Main() int {
 			sum := 0
 			values := []int{10, 11}
 			for _, v := range values {
-				binary.Itoa(v, 10)
-				sum += inline.VarSum(1, 2, 3, binary.Atoi("4", 10))
```
[the remark this answers is not in the document]
Comment (AnnaShaleva) [reply pkg/compiler/inline_test.go:143]: Replaced with `runtime.GetTime`.

File: pkg/compiler/inline_test.go [hunk 4]
```diff
@@ -118,42 +118,47 @@ func TestInline(t *testing.T) {
...
-		import "github.com/nspcc-dev/neo-go/pkg/interop/binary"
+		import "github.com/nspcc-dev/neo-go/pkg/interop/storage"
 		import "github.com/nspcc-dev/neo-go/pkg/compiler/testdata/inline"
 		func Main() int {
 			sum := 0
 			values := []int{10, 11}
 			for _, v := range values {
-				binary.Itoa(v, 10)
-				sum += inline.VarSum(1, 2, 3, 4)
+				if v != 0 { // use 'v'
```
Comment (fyrchik) [inline pkg/compiler/inline_test.go:127]: `if` creates another scope. We better just use e.g. `_ = v`.

**Comment by AnnaShaleva** [comment]
> Why are last 2 commits in this PR?

They were in neo-modules#517 along with the changes related to this PR. Removed them.

File: pkg/core/notary_test.go [hunk 5]
```diff
@@ -94,7 +94,7 @@ func TestNotary(t *testing.T) {
 	}
 
 	acc1, ntr1, mp1 := getTestNotary(t, bc, "./testdata/notary1.json", "one", onTransaction)
-	acc2, _, _ := getTestNotary(t, bc, "./testdata/notary2.json", "two", onTransaction)
+	acc2, _, _ := getTestNotary(t, bc, "./testdata/notary2.json", "one", onTransaction)
```
Comment (roman-khimov) [inline pkg/core/notary_test.go:97]: Unexpected.

File: pkg/crypto/keys/private_key_test.go [hunk 6]
```diff
@@ -14,10 +14,6 @@ import (
 func TestPrivateKey(t *testing.T) {
 	for _, testCase := range keytestcases.Arr {
 		privKey, err := NewPrivateKeyFromHex(testCase.PrivateKey)
-		if testCase.Invalid {
```
Comment (roman-khimov) [inline pkg/crypto/keys/private_key_test.go:17]: We still have invalid cases and we still need to test them, so why this is removed?

File: pkg/interop/crypto/crypto.go [hunk 7]
```diff
@@ -8,10 +8,10 @@ import (
 	"github.com/nspcc-dev/neo-go/pkg/interop/neogointernal"
 )
 
-// ECDSASecp256r1CheckMultisig checks multiple ECDSA signatures at once. It uses
-// `Neo.Crypto.CheckMultisigWithECDsaSecp256r1` syscall.
-func ECDSASecp256r1CheckMultisig(msg []byte, pubs []interop.PublicKey, sigs []interop.Signature) bool {
-	return neogointernal.Syscall3("Neo.Crypto.CheckMultisigWithECDsaSecp256r1", msg, pubs, sigs).(bool)
+// CheckMultisig checks that script-container is signed by multiple ECDSA keys
```
Comment (roman-khimov) [inline pkg/interop/crypto/crypto.go:11]: `script container (transaction)`

**Review by roman-khimov — COMMENTED** [review]
Looks like native contract coverage can be improved for erroneous cases.

File: pkg/core/notary_test.go (hunk 5 above)
Comment (AnnaShaleva) [reply pkg/core/notary_test.go]: Fixed.

**Comment by roman-khimov** [comment]
> They were in neo-modules#517 along with the changes related to this PR.

Another PR for them?

File: pkg/crypto/keys/nep2_test.go [hunk 8]
```diff
@@ -15,7 +15,6 @@ func TestNEP2Encrypt(t *testing.T) {
 			assert.Error(t, err)
 			continue
 		}
-
```
Comment (roman-khimov) [inline pkg/crypto/keys/nep2_test.go:18]: Uh-oh.

**Review by roman-khimov — APPROVED** [review]

File: pkg/crypto/keys/nep2_test.go (hunk 8 above)
Comment (AnnaShaleva) [reply pkg/crypto/keys/nep2_test.go]: These empty lines are so treacherous

**Pull request merged**